File: src/main/java/org/tensorflow/keras/activations/Lambda.java
package org.tensorflow.keras.activations;

import org.tensorflow.Operand;
import org.tensorflow.keras.mixin.ActivationFunction;
import org.tensorflow.op.Ops;
import org.tensorflow.types.family.TNumber;

/**
 * Creates an `Activation` from an unnamed function.
 * @param <T>
 */
public class Lambda<T extends TNumber> extends Activation<T> {
    private final ActivationFunction<T> activation;

    /**
     * Creates an Activation function.
     * @param unnamedActivation An activation function.
     */
    public Lambda(ActivationFunction<T> unnamedActivation) {
        super();
        this.activation = unnamedActivation;
    }

    /**
     * Applies the given activation.
     */
    @Override
    protected Operand<T> call(Ops tf, Operand<T> inputs) {
        return activation.apply(tf, inputs);
    }
}

File: src/main/scala/org/tensorflow/keras/activations/Lambda.scala
package org.tensorflow.keras.activations

import org.tensorflow.Operand
import org.tensorflow.keras.mixin.ActivationFunction
import org.tensorflow.op.Ops
import org.tensorflow.types.family.TNumber

/**
  * Creates an `Activation` from an unnamed function.
  *
  * @param unnamedActivation An activation function.
  * @param < T>
  */
class Lambda[T <: TNumber](val activation: ActivationFunction[T]) extends Activation[T] {
  /**
    * Applies the given activation.
    */
  override protected def callOne(tf: Ops, inputs: Operand[T]): Operand[T] = activation.apply(tf, inputs)
}
